components/formats-common/src/loci/common/URLHandle.java

```diff
@@ -64,8 +64,10 @@
    * Constructs a new URLHandle using the given URL.
    */
   public URLHandle(String url) throws IOException {
-    if (!url.startsWith("http") && !url.startsWith("file:")) {
-      url = "http://" + url;
+    if (!url.startsWith("https") && !url.startsWith("file:")) {
+        url = "https://" + url;
+    }else if (!url.startsWith("http") && !url.startsWith("file:")) {
+        url = "http://" + url;
```

Comment:
@mtbc has it right - the original code was meant to add a valid URL scheme (```http://```) if none is present.

Assuming URLHandle can accept a value of ```url``` in one of the following formats:

1. ```abc.com/xyz.png```
2. ```file://a/b/c/xyz.png```
3. ```http://abc.com/xyz.png```
4. ```https://abc.com/xyz.png```

the original code would only have changed the value of ```url``` in case 1 (it would become ```http://abc.com/xyz.png```).

With this change:

1. becomes ```https://abc.com/xyz.png```
2. is not changed
3. becomes ```https://http://abc.com/xyz.png```
4. is not changed